client-side/src/stories/genericInput.jsx

```diff
@@ -0,0 +1,86 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { TextField, InputAdornment } from '@mui/material';
+import '../style/genericInput.scss';
+
+const GenericInput = ({ 
+  label, 
+  type = 'text', 
+  value = '', 
+  onChange = () => {}, 
+  size = 'medium', 
+  width = '20%', 
+  icon: Icon=null, 
+  disabled= false,
+  validation  = () => {}, 
+  ...rest 
+}) => {
+  const [inputValue, setInputValue] = useState(value);
+  const [error, setError] = useState(false);
+  const [helperText, setHelperText] = useState('');
+
+  useEffect(() => {
+    if (validation && typeof validation === 'function') {
+      handleValidation(inputValue);
+    }
+  }, [inputValue]);
+
+  const handleChange = (e) => {
+    const newValue = e.target.value;
+    setInputValue(newValue);
+    onChange(newValue);
```

Comment:
please check if onChange is exists because it's not defined as isRewuired in thepropTypes object